lib/goth/token.ex

```diff
@@ -157,7 +154,7 @@ defmodule Goth.Token do
     grant_type = "urn:ietf:params:oauth:grant-type:jwt-bearer"
     body = "grant_type=#{grant_type}&assertion=#{jwt}"
 
-    response = Goth.HTTPClient.request(config.http_client, :post, url, headers, body, [])
+    response = request(config.http_client, :post, url, headers, body, [])
```

Comment:
if we do this:

```suggestion
    response = request(config.http_client, method: :post, url: url, headers: headers, body: body)
```

then I think our `request/2` is gonna be super simple:

```elixir
defp request({fun, extra_options}, options) do
  fun.(options ++ extra_options)
end
```

Btw, nitpick, please use the word `fun` instead of `func` as it is more common for this type of a thing.

Reply:
I would make a joke about that, but probably wouldn't become fun to read. 😹